From bcb7e08c52310c0e9c21826ed53911971447e4b3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:53:17 -0400
Subject: [tools] perf: Handle PERF_RECORD_HEADER_EVENT_TYPE properly

Message-id: <1408953210-26343-310-git-send-email-jolsa@redhat.com>
Patchwork-id: 89039
O-Subject: [PATCH RHEL7.1 BZ1133083 309/322] perf tools: Handle PERF_RECORD_HEADER_EVENT_TYPE properly
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f67697bd079f4bbcbe7d6d26765a06b18afe0630
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Feb 4 15:37:48 2014 +0100

description
===========
We removed event types from data file in following commits:

  6065210 perf tools: Remove event types framework completely
  44b3c57 perf tools: Remove event types from perf data file

We no longer need this information, because we can get it directly from
tracepoints.

But we still need to handle PERF_RECORD_HEADER_EVENT_TYPE event for the
sake of old perf data files created in pipe mode like:

  $ perf.3.4 record -o - foo >perf.data
  $ perf.312 report -i - < perf.data
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/session.c b/tools/perf/util/session.c
index 0b39a48..5da6ce7 100644
--- a/tools/perf/util/session.c
+++ b/tools/perf/util/session.c
@@ -1008,6 +1008,12 @@ static int perf_session__process_user_event(struct perf_session *session, union
   if (err == 0)
    perf_session__set_id_hdr_size(session);
   return err;
+ case PERF_RECORD_HEADER_EVENT_TYPE:
+  /*
+   * Depreceated, but we need to handle it for sake
+   * of old data files create in pipe mode.
+   */
+  return 0;
  case PERF_RECORD_HEADER_TRACING_DATA:
   /* setup for reading amidst mmap */
   lseek(fd, file_offset, SEEK_SET);
-- 
1.7.1